db/migrate/20151119070640_remove_dangling_entries.rb

```diff
@@ -0,0 +1,22 @@
+class RemoveDanglingEntries < ActiveRecord::Migration
+  def up
+    UserSnp.find_each do |us|
+      if Snp.find_by_name(us.snp_name).nil? || Genotype.find_by_id(us.id).nil?
+        UserSnp.destroy(us.id)
+      end
+    end
```

Comment:
When importing user-SNPs, we create SNPs if they should not already exist. When there are user-SNPs without corresponding SNPs, wouldn't that be due to faulty data? Are there cases where we would delete SNPs and want the associated user-SNPs to be gone as well? Also, did you try this locally? I'm not sure this works since UserSnps don't have an `id` column.